tests/Benchmarks/System.Text.Primitives/PrimitiveParserInt32PerfTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Primitives.Tests;
+using BenchmarkDotNet.Attributes;
+using Benchmarks.System.Text.Primitives;
+
+namespace System.Text.Primitives.Benchmarks
+{
+    public partial class PrimitiveParserPerfTests
+    {
+        private const int InnerCount = 10000;
+
+        private static readonly string[] s_Int32TextArray = new string[20]
+        {
+            "214748364",
+            "2",
+            "21474836",
+            "-21474",
+            "21474",
+            "-21",
+            "-2",
+            "214",
+            "-21474836",
+            "-214748364",
+            "2147",
+            "-2147",
+            "-214748",
+            "-2147483",
+            "214748",
+            "-2147483648",
+            "2147483647",
+            "21",
+            "2147483",
+            "-214"
```

Comment:
I think it would be worthwhile to add test cases like these, no?
"000000000000000000001235abcdfg"
"2abcdefghijklmnop"